### PR TITLE
vault-24597: add key types and key creation for CMAC

### PR DESCRIPTION
**Pull request by rculpepper**

### CODE REVIEW

File: sdk/helper/keysutil/policy.go [hunk 1]
```diff
@@ -1611,14 +1617,17 @@ func (p *Policy) RotateInMemory(randReader io.Reader) (retErr error) {
...
 	}
-	entry.HMACKey = hmacKey
 
+	var err error
 	switch p.Type {
-	case KeyType_AES128_GCM96, KeyType_AES256_GCM96, KeyType_ChaCha20_Poly1305, KeyType_HMAC:
+	case KeyType_AES128_GCM96, KeyType_AES256_GCM96, KeyType_ChaCha20_Poly1305, KeyType_HMAC, KeyType_AES128_CMAC, KeyType_AES256_CMAC:
 		// Default to 256 bit key
 		numBytes := 32
 		if p.Type == KeyType_AES128_GCM96 {
```
Comment (stevendpclark) [inline sdk/helper/keysutil/policy.go:1633]: I believe we are missing KeyType_AES128_CMAC here, so both CMAC key types will be 256 bit?

File: builtin/logical/transit/path_keys_test.go [hunk 2]
```diff
@@ -196,3 +196,113 @@ func TestTransit_CreateKeyWithAutorotation(t *testing.T) {
 		})
 	}
 }
+
+// TestTransit_CreateKey validates transit key creation functionality
+func TestTransit_CreateKey(t *testing.T) {
```
Comment (stevendpclark) [inline builtin/logical/transit/path_keys_test.go:201]: Love it!

**Review by stevendpclark — COMMENTED** [review]
This looks great! One small comment.

**Review by sgmiller — CHANGES_REQUESTED** [review]
Echoing Steve's comment about not understanding key sizes (forgot the spec on this), and one comment but otherwise lgtm

**Review by stevendpclark — APPROVED** [review]
👍

**Review by sgmiller — APPROVED** [review]

**Pull request merged**